pwnlib/util/proc.py

```diff
@@ -298,7 +322,12 @@ def wait_for_debugger(pid):
     Returns:
         None
     """
-    with log.waitfor('Waiting for debugger') as l:
-        while tracer(pid) is None:
-            time.sleep(0.01)
+    with log.waitfor('Waiting for debugger on pid %d' % pid) as l:
```

Comment:
It would be better to modify the original `tracer` method to call this instead, e.g.:

```
def tracer(pid):
    if sys.platform == 'win32':
        return tracer_win(pid)
    # old tracer logic remains in place)
```